framework/include/materials/MaterialData.h

```diff
@@ -161,6 +176,12 @@ class MaterialData
   /// Status of storage swapping (calling swap sets this to true; swapBack sets it to false)
   bool _swapped;
 
+  /// Properties having dot properties
+  std::vector<const MaterialProperty<Real> *> _current_props;
+  std::vector<const MaterialProperty<Real> *> _old_props;
+  std::vector<const MaterialProperty<Real> *> _older_props;
+  std::vector<MaterialProperty<Real> *> _dot_props;
```

Comment:
How are these different from `_props_old` above?

Reply:
They contains pointers of material properties that are only needed for dot evaluation.